### ISSUE TITLE
Response type

### ISSUE THREAD

**Issue by jacob87o2** [psrebniak/swagger-typescript-client-generator#17]
Hi,

Thanks for providing this generator, it's great!

However, I have one small issue. It assumes global Response type, which comes from browser Fetch API and which doesn't exist in node.js applications.

I think it would be better to allow generic response type. That way we can use any request factory we like, not necessarily implementing the browser Fetch API Response.

**Comment by psrebniak** [comment]
Hello
You can use isomorphic-fetch to provide fetch func and API to nodejs. 

But you’re right that more generic solution is needed. 
What solution to support for custom response types do you prefer?

**Comment by jacob87o2** [comment]
Yeah, I know there are some libraries that provide browser-like fetch method. I tried to avoid `isomorphic-fetch`, mainly because last release was like 4 years ago. I also tried `node-fetch` but it doesn't have some of the Response properties, like `formData` and `trailer`.

I don't know yet, how we could make it more generic while still maintaining typing for response objects. Let's say, we have `Something` response object. Now it's generated as `Promise<ApiResponse<Something>>` return type. 

In my case `Promise<Something>>` would be sufficient, but that would break typing if someone actually wants it to be wrapped in `Response`. Also, someone could want `AxiosResponse` for example. 

I'm in no way a typescript expert, so I don't know if it's possible to make some kind of generic `Promise<T<Something>>` return type, based on request factory passed to client constructor. Seems complicated, but I'll try to figure out something. Maybe you have some idea how it can be solved?

Meanwhile, I guess, I'll go with `isomorphic-fetch`.

**Comment by psrebniak** [comment]
@jacob87o2 next major version v2 it will be configurable by `new YourClient<ResponseType>` 
Added to milestone

**Comment by jacob87o2** [comment]
@psrebniak Awesome news! Thanks!
